Snabble/UI/Scanner/BarcodeDetectorOverlay.swift

```diff
@@ -73,8 +73,11 @@ public final class BarcodeDetectorOverlay: UIView {
         NSLayoutConstraint.activate([
             reticle.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: 16),
             reticle.trailingAnchor.constraint(equalTo: self.trailingAnchor, constant: -16),
-            reticleHeight,
-            reticle.centerYAnchor.constraint(equalTo: barcodeOverlay.centerYAnchor)
+            reticleHeight.usingPriority(.defaultHigh),
+            reticle.centerYAnchor.constraint(equalTo: barcodeOverlay.centerYAnchor),
+
+            reticle.heightAnchor.constraint(lessThanOrEqualTo: self.heightAnchor, constant: -32)
+                .usingPriority(.required - 1)
```

Comment:
```swift
            reticle.topAnchor.constraint(greaterThanOrEqualTo: topAnchor, constant: 16),
            bottomAnchor.constraint(greaterThanOrEqualTo: reticle.bottomAnchor, constant: 16),

            reticleHeight.usingPriority(.defaultHigh + 1),
            reticle.centerYAnchor.constraint(equalTo: barcodeOverlay.centerYAnchor)
```

Was hältst du davon? So ist es wirklich oben und unten an den Contraints festgemacht und kein Spielchen mit Center und Height.

Reply:
Das führt bei mir zu "width and horizontal position are ambigous" und einem 0x0 Pixel großem Reticle. Ausserdem soll `.square` ja eben nur (maximal) ein Quadrat in der Mitte sein.

Reply:
Moment, das hat bei mir funktioniert. Ich gucke es mir nochmal genauer an.